projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php

```diff
@@ -234,24 +234,46 @@ function wpcom_site_menu_enqueue_scripts() {
 
 	$notice = wpcom_get_sidebar_notice();
 	if ( $notice ) {
-		$link = $notice['link'];
+		$link = $notice->link;
 		if ( str_starts_with( $link, '/' ) ) {
 			$link = 'https://wordpress.com' . $link;
 		}
 
-		wp_localize_script(
+		$user_id    = null;
+		$user_login = null;
+
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			global $current_user;
+			$user_id    = $current_user->ID;
+			$user_login = $current_user->user_login;
+		} else {
+			$connection_manager = new Connection_Manager();
+			$wpcom_user_data    = $connection_manager->get_connected_user_data();
+			if ( $wpcom_user_data ) {
+				$user_id    = $wpcom_user_data['ID'];
+				$user_login = $wpcom_user_data['login'];
+			}
+		}
+
+		$data = (object) array(
```

Comment:
It's a bit strange that we need to cast an array to an object to call `wp_json_encode`. The only alternative I can think of is using the `JSON_FORCE_OBJECT` flag but I'm not sure if it improves anything at all tbh.

Reply:
Good catch, it's unnecessary so I removed it in https://github.com/Automattic/jetpack/pull/37214/commits/63f18b56d8526371f44fb2077e01f62f3c4bcbec